File: backend_python/src/routes/file_routes.py
from flask import Blueprint
from controllers.file_controller import upload_file, get_files_by_user, get_file_by_id, delete_file

file_bp = Blueprint('file_routes', __name__)

# Define routes on the blueprint
file_bp.route('/upload/', methods=['POST'])(upload_file)
file_bp.route('/', methods=['GET'])(get_files_by_user)
file_bp.route('/<int:file_id>', methods=['GET'])(get_file_by_id)
file_bp.route('/<int:file_id>', methods=['DELETE'])(delete_file)

def register_file_routes(app):
    # Register the blueprint with the app
    app.register_blueprint(file_bp, url_prefix='/api/files')

File: backend_nodejs/src/routes/fileRoutes.js
const express = require('express');
const fileController = require('../controllers/fileController');
const router = express.Router();
const authMiddleware = require('../middleware/authMiddleware');

router.post('/upload', authMiddleware, fileController.uploadFile);
router.get('/', authMiddleware, fileController.getFilesByUser);
router.get('/:id', authMiddleware, fileController.getFileById);
router.delete('/:id', authMiddleware, fileController.deleteFile);

module.exports = router;